src/sidebar/config/build-settings.js

```diff
@@ -76,73 +77,24 @@ function fetchConfigFromAncestorFrame(origin, window_ = window) {
 }
 
 /**
- * @deprecated
- * Merge client configuration from h service with config fetched from
- * embedding frame.
- *
- * Typically the configuration from the embedding frame is passed
- * synchronously in the query string. However it can also be retrieved from
- * an ancestor of the embedding frame. See tests for more details.
- *
- * @param {object} appConfig - Settings rendered into `app.html` by the h service.
- * @param {Window} window_ - Test seam.
- * @return {Promise<object>} - The merged settings.
- */
-function fetchConfigLegacy(appConfig, window_ = window) {
-  const hostConfig = hostPageConfig(window_);
-
-  let embedderConfig;
-  const origin = /** @type string */ (hostConfig.requestConfigFromFrame);
-  embedderConfig = fetchConfigFromAncestorFrame(origin, window_);
-
-  return embedderConfig.then(embedderConfig => {
-    const mergedConfig = Object.assign({}, appConfig, embedderConfig);
-    mergedConfig.apiUrl = getApiUrl(mergedConfig);
-    return mergedConfig;
-  });
-}
-
-/**
- * Merge client configuration from h service with config from the hash fragment.
+ * Retrieve host configuration by RPC from another frame
  *
- * @param {object} appConfig - App config settings rendered into `app.html` by the h service.
- * @param {object} hostPageConfig - App configuration specified by the embedding frame.
- * @return {object} - The merged settings.
- */
-function fetchConfigEmbed(appConfig, hostPageConfig) {
-  const mergedConfig = {
-    ...appConfig,
-    ...hostPageConfig,
-  };
-  mergedConfig.apiUrl = getApiUrl(mergedConfig);
-  return mergedConfig;
-}
-
-/**
- * Merge client configuration from h service with config fetched from
- * a parent frame asynchronously.
- *
- * Use this method to retrieve the config asynchronously from a parent
- * frame via RPC. See tests for more details.
- *
- * @param {object} appConfig - Settings rendered into `app.html` by the h service.
- * @param {Window} parentFrame - Frame to send call to.
+ * @param {Window} targetFrame - Frame to request configuration from
  * @param {string} origin - Origin filter for `window.postMessage` call.
- * @return {Promise<object>} - The merged settings.
+ * @return {Promise<HostConfig>}
  */
-async function fetchConfigRpc(appConfig, parentFrame, origin) {
+async function fetchHostConfigRPC(targetFrame, origin) {
   const remoteConfig = await postMessageJsonRpc.call(
-    parentFrame,
+    targetFrame,
     origin,
     'requestConfig',
     [],
     3000
   );
   // Closure for the RPC call to scope parentFrame and origin variables.
   const rpcCall = (method, args = [], timeout = 3000) =>
-    postMessageJsonRpc.call(parentFrame, origin, method, args, timeout);
-  const mergedConfig = fetchConfigEmbed(appConfig, remoteConfig);
```

Comment:
These changes separate out finding/getting the correct host configuration from merging configuration objects together.